api/db.py

```diff
@@ -103,10 +108,18 @@ def _convert_int_values(cls, attributes):
             if isinstance(val, tuple) and len(val) == 2 and val[0] == 'int'
         }
 
+    def _convert_bool_values(self, attributes):
+        return {
+            key: self.BOOL_VALUE_MAP[val.lower()]
+            for key, val in attributes.items()
+            if isinstance(val, str) and val.lower() in self.BOOL_VALUE_MAP
+        }
```

Comment:
It's not the optimal way to implement it as it's converting the value to lower-case twice and doing a lookup in BOOL_VALUE_MAP twice - but that's not a big deal.  I think it could be improved with either a nested comprehension or a plain for-loop e.g.:
```python
for key, val in attributes.items():
    val_bool = self.BOOL_VALUE_MAP.get(val.lower())
    if val_bool is not None:
        ret[key] = val_bool
```
Also I believe all the attributes should be strings, or is FastAPI converting some values to other types?  The `_convert_int_values()` function is there specifically for this kind of reason.

Reply:
> It's not the optimal way to implement it as it's converting the value to lower-case twice and doing a lookup in BOOL_VALUE_MAP twice

Fixed it.